x/staking/types/msg_test.go

```diff
@@ -14,6 +17,25 @@ var (
 	coinZero = sdk.NewInt64Coin(sdk.DefaultBondDenom, 0)
 )
 
+func TestMsgPkDecode(t *testing.T) {
+	// description := Description{}
+	// commission1 := NewCommissionRates(sdk.ZeroDec(), sdk.ZeroDec(), sdk.ZeroDec())
+	// msg, err := NewMsgCreateValidator(valAddr1, pk1, coinPos, description, commission1, sdk.OneInt())
+	// require.NoError(t, err)
+
+	registry := codectypes.NewInterfaceRegistry()
+	cdc := codec.NewProtoCodec(registry)
```

Comment:
you need to `RegisterInterface`  and `RegisterImplementations` on the `registry` (eg using `cryptocodec.RegisterInterfaces(registry)` in your particular case)